### DISCUSSION TITLE
Replacement for camera access view ViewState3D in iTwin V3

### DISCUSSION THREAD

**Discussion by oussamabonnor1** [iTwin/itwinjs-core#3000] [Q&A] [answered]
We're using the [Screen Space Effects](https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=screen-space-effects-sample&imodel=Villa) code example to run a **Distortion Effect** on our iModels.

The effect uses a vertex shader that needs a height property to calculate the new location of pixels, this height value is obtained by first calculating the FoV & then using it to find the height value, as shown below:

`const fov = context.viewport.view.camera.lens.radians;`
`height = Math.tan(fov / 2) / context.viewport.viewRect.aspect;`

the `view` in the viewport is a ViewState property which no longer contains the camera object as of V3.
Where has it been moved to? Or, is there a new best practice to get the lens degrees in radians now ?

P.s: We have searched high & low for it in the [NextVersion.md](https://github.com/iTwin/itwinjs-core/blob/master/docs/changehistory/NextVersion.md#update-guide) guide and all we found was a new implementation of the `lookAt` function.

**Comment by pmconne** [comment, marked as answer]
ViewState didn't have a `camera` property in V2 either (or ever, as far as I can tell).
[ViewState3d](https://www.itwinjs.org/reference/imodeljs-frontend/views/viewstate3d/) does. Camera only makes sense in 3d.
```
if (context.viewport.view.is3d() && context.viewport.view.isCameraOn) {
  const fov = context.viewport.view.camera.lens.radians;
  height = Math.tan(fov / 2) / context.viewport.viewRect.aspect;
}
```

**Reply by oussamabonnor1** [reply]
Thank you @pmconne, I was missing `context.viewport.view.is3d()` which apparently dictates which viewState is used inside the condition:

![image](https://user-images.githubusercontent.com/17766221/148549149-d669ccdf-700c-419f-82b8-a0f53fdc3141.png)

VS 

![image](https://user-images.githubusercontent.com/17766221/148549207-8f1986f7-dd9a-4c51-86de-e64245705498.png)
